protobuf/benchmarks/compile-js.sh

```diff
@@ -10,7 +10,6 @@ OUTPUT_DIR="${SCRIPT_DIR}/temp"
 
 mkdir -p "${OUTPUT_DIR}"
 
-dart2js --trust-type-annotations        \
-        --trust-primitives              \
+dart2js -O4                             \
```

Comment:
We should update this to use `dart compile js`, [dart2js documentation](https://dart.dev/tools/dart2js) says `dart2js` is deprecated:

> Version note: The dart2js command-line tool is deprecated as of Dart 2.17 and will be removed in a future release.
>
> Use [dart compile js](https://dart.dev/tools/dart-compile#js) or the [webdev](https://dart.dev/tools/webdev) tool to compile Dart code to deployable JavaScript.